plone/app/layout/globals/layout.py

```diff
@@ -93,40 +93,27 @@ def have_portlets(self, manager_name, view=None):
 
         return renderer.visible
 
-    @memoize
-    def icons_visible(self):
-        """Returns True if icons should be shown or False otherwise."""
-        context = self.context
-        membership = getToolByName(context, "portal_membership")
-        anon = membership.isAnonymousUser()
-
+    def _image_visibility(self, name):
+        """check if image {name} is visible with current settings and user"""
         registry = getUtility(IRegistry)
         settings = registry.forInterface(ISiteSchema, prefix="plone", check=False)
-        icon_visibility = settings.icon_visibility
-
-        if icon_visibility == "enabled":
-            return True
-        elif icon_visibility == "authenticated" and not anon:
+        visibility = getattr(settings, f"{name}_visibility")
```

Comment:
No default is passed to `getattr`.
Is this intentional?
If no default is passed and the attribute is not in settings we will get an `AttributeError`.
If that is the wanted behavior, ignore this comment.

Reply:
As before, as it was `settings.icon_visibility`, there was no default. It must be there anyway.